### PR TITLE
Allow resource file publish dates to be undecided.

### PR DESCRIPTION
**Pull request by spark008**

### CODE REVIEW

File: app/models/assignment_file.rb [hunk 1]
```diff
@@ -26,14 +26,36 @@ class AssignmentFile < ActiveRecord::Base
...
 
   # The time when this file will be visible to students.
-  validates :published_at, presence: true, timeliness: true
+  validates :published_at, timeliness: { type: :datetime, allow_nil: true }
+
+  # Nullify :published_at if the author did not decide a release date.
+  def act_on_reset_published_at
+    if @reset_publish_date
+      self.published_at = nil
+      @reset_publish_date = nil
```
Comment (pwnall) [inline app/models/assignment_file.rb]: Let's switch to `reset_published_at`

File: app/models/assignment_file.rb [hunk 2]
```diff
@@ -26,14 +26,36 @@ class AssignmentFile < ActiveRecord::Base
...
+  # True if the publish date was omitted (reset to nil) (virtual attribute).
+  def reset_publish_date
+    published_at.nil?
+  end
+
+  # Store the user's decision to set or omit (reset) the publish date.
+  #
+  # @param [String] state '0' if setting a date, '1' if omitting
+  def reset_publish_date=(state)
+    @reset_publish_date = state.to_i != 0
```
Comment (pwnall) [inline app/models/assignment_file.rb]: You should use `ActiveRecord::Type::Boolean.new.cast(state)` here.

Comment (spark008) [inline app/models/assignment_file.rb]: kewl

**Comment by pwnall** [comment]
Thank you!

**Pull request merged**